### DISCUSSION TITLE
Seeing `set +o history` when I record every session

### DISCUSSION THREAD

**Discussion by jefftriplett** [charmbracelet/vhs#169] [Q&A] [answered]
When I record a new VHS session, I see `set +o history; unset PROMPT_COMMAND; export PS1="%s"; clear;` as the first thing printed to my terminal. I assume this isn't by design, but I'm unsure how to avoid it. 

It appears to be from: 
https://github.com/charmbracelet/vhs/blob/6f7cc819032319a4bbef579d82c41b66bc0c427a/shell.go#L23
 
<img width="895" alt="Xnapper-2022-11-22-07 51 10-DE5B90A5-FE11-4C6C-8DC6-28884338EC7F" src="https://user-images.githubusercontent.com/50527/203330918-68977eae-4d5c-4915-96cd-605c6848d83b.png">

Any pointers in how to clear this or avoid it? I double-checked that I didn't have DEBUG set in my environment, and I'm not sure what else to look for. 

Or is this normal, and everyone crops the first 1/2 second of every recording?

**Comment by caarlos0** [comment]
are you running from `main` or the latest release? I think this was fixed already...

**Reply by jefftriplett** [reply]
Thank you for the reply. I'm running 0.1.1 from homebrew. 

https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/vhs.rb

**Reply by caarlos0** [reply]
yes, this should be fixed on main... You can install it with:

```sh
brew install vhs --HEAD
```

**Reply by jefftriplett** [reply]
I installed the HEAD version:

```
❯ vhs --version
vhs version HEAD-6f7cc81
```
Sadly, I'm still seeing the set history, etc, messages at the beginning. 

I'm happy to help troubleshoot it or to give it some time and hope that a future release might fix it.

**Reply by kranurag7** [reply]
I'm facing the same issue with the latest release. However, I'm using this to hide the error message that's coming while executing `vhs < demo.tape` 
```
Hide

Ctrl+L
Show

Sleep 2
Type "echo 'Hello!'"
Enter
```
---
```
vhs --version 
vhs version v0.2.0 (6f7cc81)
```

**Reply by jefftriplett** [reply]
I found a similiar workaround. In case it's helpful, the `Sleep` inside the `Hide/Show` block seems to be the only thing that suppresses the prompt. 

```
Hide
Type "clear"
Enter
Sleep 1s
Show
```

----
```shell
❯ vhs --version
vhs version HEAD-906c420
```

**Comment by jefftriplett** [comment]
Just following back up since it's been a while. This issue still is an issue. If I use my workaround I can clip the frames and probably rebuild with ffmpeg. 

I wonder if you have ever thought about a feature let us turn off and on recording? that might be nice so that we could clear the screen and then start taking screenshots/frames vs. always recording from the start? That might be useful for other things like doing really long downloads in a demo. 

I know "Hide/Show" sort-of does this bug if it were possible to tell vhs to hide by default, that might make it easier to work around.

**Reply by caarlos0** [reply]
have you tried it on main?

**Reply by jefftriplett** [reply]
Yes, I'm on the latest/HEAD/sha version (https://github.com/charmbracelet/vhs/commit/97c0d858e47415b168e65e4c5849bfa41ac2a382)

See screenshot: 

![demo](https://user-images.githubusercontent.com/50527/214949399-509678ee-b64b-42cd-b3a7-47a1a7f367e1.gif)

It's a head-scratcher.

**Reply by caarlos0** [reply]
ahh, I thought [this](https://github.com/charmbracelet/vhs/pull/214) was already merged

could you maybe try on that branch?

**Reply by jefftriplett** [reply]
🎉  I think it's working from your branch. I see clean output and no extra frames:

![demo](https://user-images.githubusercontent.com/50527/215517976-79e50930-f4d2-40e1-82b9-3f058c016c95.gif)

from this `demo.tape`:

```elixir
# Where should we write the GIF?
Output build/demo.gif

# Set up a 1200x600 terminal with 46px font.
Set FontSize 46
Set Width 1200
Set Height 600

# Type a command in the terminal.
Type "./vhs-git/vhs --version"

# Pause for dramatic effect...
Sleep 500ms

# Run the command by pressing enter.
Enter

# Admire the output for a bit.
Sleep 5s
```

Thank you for figuring this out. I'm excited to `brew install` once it's ready and it just work. 🎉

**Comment by jefftriplett** [comment, marked as answer]
I am happy to say this is fixed in https://github.com/charmbracelet/vhs/pull/214
